src/ape/api/providers.py

```diff
@@ -802,12 +804,14 @@ def _get_logs_in_block_range(
                             f"'{name}' is not an indexed topic for event '{abi.name}'."
                         )
 
-                    search_topics.append(arg)
+                    search_topic_values.append(arg)
                     abi_types.append(abi_type)
 
                 encoded_topic_data = [
-                    encode_single(topic_type, topic_data).hex()  # type: ignore
-                    for topic_type, topic_data in zip(topics.types, search_topics)
+                    to_hex(keccak(encode_single_packed(abi_type, value)))
```

Comment:
@banteg Changed it from `encode_hex` to `to_hex`, does this still work for you?

(also handles that weird Solidity feature with dynamic indexed queries that no one uses)